blocks/README.md

```diff
@@ -154,6 +155,43 @@ Returns settings associated with a registered block.
 
 Returns settings associated with a registered control.
 
+## Backwards Compatibility
+
+When defining a block, you can choose to associate its compatibility with content shapes that had existed prior to the introduction of blocks. A block's `compatibility` property can include one or more objects which describe how the block can be initialized in cases where the editor cannot find another applicable handler.
+
+For example, consider a paragraph of text. A post's content may contain paragraphs, either explicitly with a paragraph `<p>` tag or inferred by newlines courtesy of the default [`wpautop` behavior](https://developer.wordpress.org/reference/functions/wpautop/). Prior to block parsing, content is normalized to apply `wpautop`, so thankfully we need only deal with the paragraph elements.
+
+We want to define a text block to enable controls specific to text (alignment, etc.), but we also want to apply this treatment to paragraph tags which were not saved with block demarcations.
+
+To do so, we'll declare compatibility with the paragraph tag when registering the block:
+
+```js
+registerBlock( 'core/text', {
+	attributes: {
+		content: html( 'p' )
+	},
+
+	compatibility: [
+		{
+			tagName: 'p'
```

Comment:
Is it that text compatibility is a poor example here, and instead unparseable text should simply fall back to the freeform block type? In which case, we should clarify the role of the freeform block vs. the text block.

Later in the documentation is consideration for when a block compatibility initializer needs to bail on an unsupportable type.

I agree it's less clear on how to handle compatibility matched elements which contain descendant matchable blocks. Might tie into discussion of support for nested blocks?